prereise/gather/demanddata/transportation_electrification/data_process.py

```diff
@@ -0,0 +1,133 @@
+import pandas as pd
+
+# Download CSV of NHTS 2017 from https://nhts.ornl.gov/
+
+
+def load_data(census_division: int, filepath: str = "trippub.csv"):
+    """Load the data at trippub.csv.
+
+    :param int census_division: any of the 9 census regions defined by the US census bureau
+    :param str filepath: filepath for the "trippub.csv" file in downloaded NHTS data
+    :raises ValueError: if the census division is not between 1 and 9, inclusive.
+    :return: (*pd.DataFrame*) -- the data loaded from trippub.csv.
+    """
+    if not (1 <= census_division <= 9):
+        raise ValueError("census_region must be between 1 and 9 (inclusive).")
+
+    nhts_census = pd.read_csv(filepath)
+
+    df = pd.DataFrame(nhts_census)
+
+    return df
+
+
+def calculate_dwell_time(data: pd.DataFrame):
+    """Calculates the dwell time, how long a vehicle has been charging
+
+    :param pd.DataFrame data: the data to calculate the dwell time from
+    :return: (*pd.Series*) -- list of dwell times
+    """
+    dwells = (
+        data["Start time (hour decimal)"].iloc[1:].values
+        - data["End time (hour decimal)"].iloc[:-1]
+    )
+    dwells.loc[data.index[-1]] = (
+        24
+        - data["End time (hour decimal)"].iloc[-1]
+        + data["Start time (hour decimal)"].iloc[0]
+    )
+
+    return dwells
+
+
+def data_filtering(census_division):
+    """Filter raw NHTS data to be used in mileage.py
+
+    :param int census_division: any of the 9 census regions defined by the US census bureau
+    :param str filepath: filepath for the "trippub.csv" file in downloaded NHTS data
+    :return: (*pd.DataFrame*) -- filtered and sorted trip data
+    """
```

Comment:
Since this is the main user-facing function, we want to make sure that the docstring here has the right input parameters listed.